backend/dataall/modules/datasets/indexers/dataset_indexer.py

```diff
@@ -34,7 +36,7 @@ def upsert(cls, session, dataset_uri: str):
                     'source': dataset.S3BucketName,
                     'resourceKind': 'dataset',
                     'description': dataset.description,
-                    'classification': dataset.confidentiality,
+                    'classification': re.sub('[^A-Za-z0-9]+', '', dataset.confidentiality),
```

Comment:
Adding this to update the confidentiality to remove any special characters/spaces